protocol/x/subaccounts/keeper/transfer.go

```diff
@@ -209,14 +209,77 @@ func (k Keeper) WithdrawFundsFromSubaccountToAccount(
 	)
 }
 
-// TransferFeesToFeeCollectorModule translates the assetId and quantums into a sdk.Coin,
-// and moves the funds from subaccounts module to the `fee_collector` module account by calling
-// bankKeeper.SendCoins(). Does not change any individual subaccount state.
-func (k Keeper) TransferFeesToFeeCollectorModule(
+// DistributeFees calculates the market mapper revenue share and fee collector share
+// based on the quantums and perpetual parameters, and transfers the fees to the
+// market mapper and fee collector.
+func (k Keeper) DistributeFees(
 	ctx sdk.Context,
 	assetId uint32,
 	quantums *big.Int,
 	perpetualId uint32,
+) error {
+	// get perpetual
+	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpetualId)
+	if err != nil {
+		return err
+	}
+
+	collateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, perpetualId)
+	if err != nil {
+		return err
+	}
+
+	// calculate market mapper rev share
+	marketMapperShare := big.NewInt(0)
+	revShareAddr, revSharePpm, err := k.revShareKeeper.GetMarketMapperRevenueShareForMarket(
```

Comment:
nit: return `err` if it's non-nil?

Reply:
This was done intentionally. The likelihood of getting an error from this function is really low - only happens if something has gone very wrong in setting params at inception.
I'd prefer not to fail trades if revshare is failing (at least initially) and use monitoring to catch this instead

Reply:
Can you add a comment to make this explicit, and add an explicit error log?